django/template/base.py

```diff
@@ -927,6 +945,27 @@ def render_annotated(self, context):
                 e.template_debug = context.template.get_exception_info(e, self.token)
             raise
 
+    def stream(self, context):
+        """
+        Return a string generator rendering the node.
+        """
+        yield self.render(context)
+
+    def stream_annotated(self, context):
+        """
+        Render the node. If debug is True and an exception occurs during
+        rendering, the exception is annotated with contextual line information
+        where it occurred in the template. For internal usage this method is
+        preferred over using the render method directly.
+        """
+        try:
+            for chunk in self.stream(context):
+                yield chunk
+        except Exception as e:
+            if context.template.engine.debug and not hasattr(e, 'template_debug'):
+                e.template_debug = context.template.get_exception_info(e, self.token)
+            raise
```

Comment:
Could you add this code path to be tested in ``template_tests.test_nodelist.ErrorIndexTest`? That's where we make sure the debug info is added correctly when calling ``template.render``.